### DISCUSSION TITLE
JVM settings seem to be ignored

### DISCUSSION THREAD

**Discussion by mrvdcsg** [Opster/opensearch-k8s-operator#651] [Q&A] [answered]
I am using configuration for my nodepools to set the JVM heap to use 2 gigs of memory but that seems to be ignored.
jvm: -Xmx2048M -Xms2048M
When I connect to the pods, the service is only using 1gig no matter what I set this to. Any idea why this is being ignored and how I can increase my heap size? I have attached part of our config for reference. Thanks!
  nodePools:
    - component: master
      replicas: 3
      diskSize: "100Gi" # The disk size to use 
      jvm: -Xmx2048M -Xms2048M
      resources: 
        requests:
          memory: "3Gi"
          cpu: "1000m"
        limits:
          memory: "3Gi"
          cpu: "1000m"

98 org.opensearch.bootstrap.OpenSearch -Enetwork.publish_host=opensearch-cluster-master-1 -Ecluster.initial_master_nodes=opensearch-cluster-bootstrap-0 -Ecluster.name=opensearch-cluster -Ehttp.port=9200 -Ediscovery.seed_hosts=opensearch-cluster-discovery -Enetwork.bind_host=0.0.0.0 -Enode.roles=cluster_manager,data,remote_cluster_client,ingest -Xshare:auto -Dopensearch.networkaddress.cache.ttl=60 -Dopensearch.networkaddress.cache.negative.ttl=10 -XX:+AlwaysPreTouch -Xss1m -Djava.awt.headless=true -Dfile.encoding=UTF-8 -Djna.nosys=true -XX:-OmitStackTraceInFastThrow -XX:+ShowCodeDetailsInExceptionMessages -Dio.netty.noUnsafe=true -Dio.netty.noKeySetOptimization=true -Dio.netty.recycler.maxCapacityPerThread=0 -Dio.netty.allocator.numDirectArenas=0 -Dlog4j.shutdownHookEnabled=false -Dlog4j2.disable.jmx=true -Djava.locale.providers=SPI,COMPAT **-Xms1g -Xmx1g** -XX:+UseG1GC -XX:G1ReservePercent=25 -XX:InitiatingHeapOccupancyPercent=30 -Djava.io.tmpdir=/tmp/opensearch-13888664803975905135 -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=data -XX:ErrorFile=logs/hs_err_pid%p.log -Xlog:gc*,gc+age=trace,safepoint:file=logs/gc.log:utctime,pid,tags:filecount=32,filesize=64m -Dclk.tck=100 -Djdk.attach.allowAttachSelf=true -Djava.security.policy=/usr/share/opensearch/config/opensearch-performance-analyzer/opensearch_security.policy --add-opens=jdk.attach/sun.tools.a

**Comment by swoehrl-mw** [comment, marked as answer]
Hi @mrvdcsg. How are you checking the heap?
When trying to reproduce this I found that the log that prints the list of JVM arguments has some defaults for `Xms` and `Xmx` in the list but near the end were the correct overrides I put in.
Also checking with `curl -k -u admin:admin https://localhost:9200/_nodes/jvm?pretty | grep heap_max_in_bytes` showed the correct heap after the cluster was bootstrapped (note the bootstrap pod has a different heap configuration).

**Reply by mrvdcsg** [reply]
I was logging into the pod and checking the actively running java jps -lvm | grep Xms. I have confirmed that Opensearch is listed with the values expected when I do the call from Opensearch. I assume it is taking advantage of that and using the memory I specify. I was concerned when it was showing 1G. Performance is fine at this point though we haven't put much on it so I'll continue to monitor performance as we go. Thanks for the reply!
